File: Data Structures & Algorithms/longest-consecutive-sequence/submission-3.kt
class Solution {
    fun longestConsecutive(nums: IntArray): Int {
        var numSet = mutableSetOf<Int>()
        for (num in nums) {
            numSet.add(num)
        }
        var output = 0
        for (num in nums) {
            if (numSet.contains(num - 1)) {
                continue
            }
            var count = 1
            var current = num
            while (numSet.contains(current + 1)) {
                current += 1
                count += 1
            }
            output = maxOf(output, count)
        }
        return output
    }
}

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-3.swift
class Solution {
    func productExceptSelf(_ nums: [Int]) -> [Int] {

        var prefix = Array(repeating: 1, count: nums.count)
        var suffix = Array(repeating: 1, count: nums.count)

        var current = 1
        for i in 0 ..< nums.count {
            prefix[i] = current
            current *= nums[i]
        }

        current = 1
        for i in (0 ..< nums.count).reversed() {
            suffix[i] = current
            current *= nums[i]
        }

        var output = Array(repeating: 1, count: nums.count)
        for i in 0 ..< nums.count {
            output[i] = suffix[i] * prefix[i]
        }

        return output

    }
}
